pkgs/build-support/rust/fetchcargo.nix

```diff
@@ -1,8 +1,24 @@
-{ stdenv, cacert, git, rust, cargo-vendor }:
+{ stdenv, cacert, git, rust, cargo-vendor, python3 }:
+let cargo-vendor-normalise = stdenv.mkDerivation {
+  name = "cargo-vendor-normalise";
+  src = ./cargo-vendor-normalise.py;
+  unpackPhase = ":";
+  installPhase = "install -D $src $out/bin/cargo-vendor-normalise";
+  doInstallCheck = true;
+  installCheckPhase = ''
+    # check that ./fetchcargo-default-config.toml is a fix point
+    reference=${./fetchcargo-default-config.toml}
+    < $reference $out/bin/cargo-vendor-normalise > test;
+    cmp test $reference
+  '';
+  buildInputs = [ (python3.withPackages(ps: [ ps.toml ])) ];
```

Comment:
I just realized we cannot use the results `python3.withPackages` to replace the shebang of the shell script on macOS, instead we must call the wrapped python executable and pass `cargo-vendor-normalise.py` as the first argument. 
On macOS shebangs must be binaries and cannot be scripts with shebangs itself.

Reply:
this is what @Infinisil got on his machine: https://hastebin.com/uxihohicax.sql